src/Features/Core/Portable/Shared/Extensions/DocumentExtensions.cs

```diff
@@ -15,8 +14,8 @@ internal static class DocumentExtensions
     {
         public static bool ShouldHideAdvancedMembers(this Document document)
         {
-            var service = document.Project.Solution.Workspace.Services.GetService<IOptionService>();
-            return service.GetOption(CompletionOptions.HideAdvancedMembers, document.Project.Language);
+            return document.Project.Solution.Workspace.Options
```

Comment:
Not directly related to your change, but this is a little scary.  You are passing around potentially changed options in your context, but this goes and gets the unchanged options from the workspace.

Luckily, this isn't an option you change, but still a little iffy.